elastic-apm-agent-premain/src/main/java/co/elastic/apm/agent/premain/AgentMain.java

```diff
@@ -34,9 +37,14 @@
  */
 public class AgentMain {
 
+    private static final AgentCracResource agentCracResource = new AgentCracResource();
```

Comment:
Is there a reason to hold a reference to the object? It seems like it only needs to be registered, it could be created then?

Reply:
@jackshirazi , Hi.
According to AWS documentation we need to have a strong reference for Crac resource:
https://docs.aws.amazon.com/lambda/latest/dg/snapstart-runtime-hooks.html

Thanks

Reply:
Thanks, I missed that